### ISSUE TITLE
Cannot change the type of a column

### ISSUE THREAD

**Issue by vipera7** [metabase/metabase#48662]
### Describe the bug

Cannot change the type of a column.

### To Reproduce

1. Navigate to the "Upload Data" section.
2. Select and upload a CSV file.
_Sample data:_

```
Date;SerialNumber0;RevisionID
11/10/2024 18:47:01;48211;1
11/10/2024 18:47:02;48212;0

```
A model is automatically created.
Modify the CSV file as follows:

```
Date;SerialNumber0;RevisionID
11/10/2024 18:47:01;48211;1
11/10/2024 18:47:02;48212;3
```
3. Click on "Replace all data in this model" using the updated CSV file.
4. Encounter the following error during upload:

```
Upload error details:
There were some errors while uploading test_upload.csv:
'3' is not a recognizable boolean
```

Problem: I am unable to change the data type of the "RevisionID" column using the metadata.

### Expected behavior

I should have the ability to modify the data type of this column to upload the updated file.

### Logs

[8ccb5d5a-3660-47ff-bdee-60ae91bdece5] 2024-10-14T10:58:54+02:00 DEBUG metabase.server.middleware.log GET /api/setup/admin_checklist 200 73,9 ms (11 DB calls) App DB connections: 0/15 Jetty threads: 4/50 (5 idle, 0 queued) (85 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[8ccb5d5a-3660-47ff-bdee-60ae91bdece5] 2024-10-14T10:58:55+02:00 DEBUG metabase.server.middleware.log GET /api/table/2838 200 20,2 ms (3 DB calls) App DB connections: 1/15 Jetty threads: 5/50 (6 idle, 0 queued) (85 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 2}
[8ccb5d5a-3660-47ff-bdee-60ae91bdece5] 2024-10-14T10:58:56+02:00 DEBUG metabase.server.middleware.log GET /api/persist 200 50,2 ms (5 DB calls) App DB connections: 0/15 Jetty threads: 5/50 (6 idle, 0 queued) (85 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id 1}
[8ccb5d5a-3660-47ff-bdee-60ae91bdece5] 2024-10-14T10:58:57+02:00 WARN metabase.server.middleware.log POST /api/table/2838/replace-csv 422 1,5 s (7 DB calls) {:metabase-user-id 2} 
{:message "'3' is not a recognizable boolean"}

### Information about your Metabase installation

Metabase version: v0.50.29
Metabase hosting environment: Jar-file
Metabase internal database : MySQL

### Severity

Cannot upload new data to the model

**Comment by crisptrutski** [comment]
This would be fixed if we allow boolean columns to be promoted to integers @luizarakaki

**Comment by luizarakaki** [comment]
Yes, we should support this type of resolution. We already do with some, right? Like int > float.
Makes sense to do boolean > int or boolean > float.
